### PR TITLE
Parsemode

### PR DESCRIPTION
**Pull request by joakim-hove**
With the changes in the parser here: https://github.com/OPM/opm-parser/pull/527 there have been some minor changes to OPM-Data to make sure all the decks pass in the strictest mode.

It is of course possible to debate how strict we should be; please read through the commit messages and voice your opinion.

### CODE REVIEW

**Comment by atgeirr** [comment]
I am ok with making data files we author ourselves as good as possible, using as strict interpretations as we can manage of everything. However, the Norne case is not such a case, on the contrary it is a test that we are able to handle not-quite-well-written (in deck terms) cases from other people. Therefore I do not think we should modify it in general. In this case it is just a stray slash, but my point is that the file as distributed by NTNU contains that slash, and we should be able to use it as distributed.

**Comment by joakim-hove** [comment]
> However, the Norne case is not such a case

I saw that one coming :-). As I see it these are the options:
1. Change the Norne deck as suggested in this PR.
2. Set the default value of `ParseMode::randomSlash` to `InputError::IGNORE`
3. In flow we can do:

```
  ParseMode parseMode;
  parseMode.randomSlash = InputError::IGNORE; 
  parser.parseFile( input_file , parseMode );
```

This again should be controlled by the properties system of `flow`.

My suggestion is to go with 3?

**Comment by atgeirr** [comment]
> My suggestion is to go with 3?

I totally agree. Can you do this?

**Comment by alfbr** [comment]
I support going with 3. I believe strictness level should be set by the module using parser. Generally speaking we should avoid changing the Norne deck unless we see compelling reasons in favour. Simply because it already is a well known reference deck. For the SPE cases it is different. Here we aim to provide reference decks for wide usage, they should be correct. Tests of incorrect decks should be handled separately. I do feel ownership to all new SPE decks, so please allow me some time to go through the proposed changes (I am currently on vacation). Thanks for the fixes!

**Comment by joakim-hove** [comment]
> Can you do this?

Yes - I have now updated this PR to not touch the Norne deck, and https://github.com/OPM/opm-autodiff/pull/421 has been updated to set `parsemode.randomSlash = InputError::IGNORE` unconditionally.

**Comment by joakim-hove** [comment]
> I do feel ownership to all new SPE decks, so please allow me some time to go through the proposed changes

Sure - this will linger here until you give it a thumbs up.

**Comment by alfbr** [comment]
From what I can see EQLDIMS is not mandatory for these cases, why do you want to include it in the deck?

Actually, EQLDIMS looks like one of the keywords from the days without dynamic memory allocation, so a closer inspection might even conclude that no OPM modules would ever use it.

The fixes on the old SPE cases are clear improvements.

Hence, I suggest leaving EQLDIMS out of the decks, and merging the other fixes.

**Comment by alfbr** [comment]
Sorry, vacation makes my brain slow. Forgot about TABDIMS, same question applies there as with EQLDIMS, why should we include it?

**Comment by joakim-hove** [comment]
> Actually, EQLDIMS looks like one of the keywords from the days without dynamic memory allocation, so a closer inspection might even conclude that no OPM modules would ever use it.

I guess you are right that the `EQLDIMS` and `TABDIMS` keywords were born in the time before dynamic memory allocation, that part is clearly irrelevant now. However the `xxxxDMIS` keywords are also used to assist during the parse process  - for instance the relationship between `EQLDIMS` and `EQUIL` is as follows:

``` txt
EQLDIMS
  3   /          

....                                                 

EQUIL                                            
  1    2    3    4 ... / -- Region 1      
  1    5    6    7 ..../ -- Region 2
  3    5    7    8 ..../ -- Region 3
```

I.e. the '3' in EQLDIMS says that the `EQUIL` keyword will contain exactly three records; now the crucial point is that _since we know that the `EQUIL` keyword will contain three records it is not slash terminated_.

To summarize:
1. The ECLIPSE documentation states that the `xxxDIMS` keywords are used to determine the   number of saturation tables/equilibration regions/... 
2. The parser makes use of the values found in the `xxxDIMS` keywords during the parse process.

So all in all I would argue that these keywords should be present in decks we create ourselves to be as correct as possible.

**Comment by alfbr** [comment]
Thanks for the explanations! Thanks also for informing me about the reference to EQLDIMS in the EQUIL keyword. Merging.

**Comment by joakim-hove** [comment]
Thanks for merging.
